src/lib/comms/include/sol-http-server.h

```diff
@@ -208,6 +218,50 @@ int sol_http_server_set_last_modified(struct sol_http_server *server, const char
 int sol_http_server_send_response(struct sol_http_request *request, struct sol_http_response *response);
 
 /**
+ * @brief Send the response and keep connection alive to request given in the
+ * callback registered on @c sol_http_server_register_handler.
+ *
+ * After this call, the caller will be resposible to close the connection calling
+ * @c sol_http_response_sse_del. @note all the necessary headers are set by this function.
+ *
+ * @param request The request given on the callback of
+ * @c sol_http_server_register_handler.
+ * @param cb Callback called when the connection is closed from the client side.
+ * @param cb_data The data pointer to be passed to @a cb.
+ *
+ * @return @c sol_http_response_sse on success, @c NULL otherwise.
+ *
+ * @see sol_http_response_sse_del
+ * @see sol_http_response_sse_send_data
+ */
+struct sol_http_response_sse *sol_http_server_send_response_sse(struct sol_http_request *request,
```

Comment:
I will rename to `on_del` because microhttpd calls the internal callback when the socket is deleted and not closed. Also the function exposed to the user is `sol_http_response_progressive_del`